apple/RNGestureHandlerModule.mm

```diff
@@ -140,18 +144,36 @@ - (void)setBridge:(RCTBridge *)bridge
 #endif // RCT_NEW_ARCH_ENABLED
 
 #ifdef RCT_NEW_ARCH_ENABLED
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(install)
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(installUIRuntimeBindings)
 {
-  dispatch_block_t block = ^{
-    RCTCxxBridge *cxxBridge = (RCTCxxBridge *)self.bridge;
-    auto runtime = (jsi::Runtime *)cxxBridge.runtime;
-    decorateRuntime(*runtime);
-  };
-  if (_dispatchToJSThread) {
-    _dispatchToJSThread(block);
-  } else {
-    [self.bridge dispatchBlock:block queue:RCTJSThread];
-  }
+  jsi::Runtime &rt = *_rnRuntime;
+
+  const auto arrayBufferValue =
+      rt.global().getProperty(rt, "_WORKLET_RUNTIME").getObject(rt).getArrayBuffer(rt).data(rt);
+  const auto uiRuntimeAddress = reinterpret_cast<uintptr_t *>(&arrayBufferValue[0]);
+  jsi::Runtime &uiRuntime = *reinterpret_cast<jsi::Runtime *>(*uiRuntimeAddress);
+
+  __weak RNGestureHandlerModule *weakSelf = self;
+
+  auto setGestureStateNew = jsi::Function::createFromHostFunction(
+      uiRuntime,
+      jsi::PropNameID::forAscii(uiRuntime, "_setGestureStateNew"),
+      2,
+      [weakSelf](jsi::Runtime &runtime, const jsi::Value &, const jsi::Value *args, size_t count) -> jsi::Value {
+        if (count == 2) {
+          const auto handlerTag = static_cast<int>(args[0].asNumber());
+          const auto state = static_cast<int>(args[1].asNumber());
+
+          // TODO: expose to JS and dispatch to UI thread if called on JS?
+          RNGestureHandlerModule *strongSelf = weakSelf;
+          if (strongSelf != nullptr) {
+            [strongSelf setGestureState:state forHandler:handlerTag];
+          }
+        }
+        return jsi::Value::undefined();
+      });
+
+  uiRuntime.global().setProperty(uiRuntime, "_setGestureStateNew", std::move(setGestureStateNew));
```

Comment:
Why don't we move this code to the same place where `isViewFlatteningDisabled` is defined?

Reply:
It needs to be delayed since there's no guarantee UI runtime will be initialized and injected into the JS one by this point.